src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function summator(...arr) {
+    if (arr.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += arr.reduce((a, b) => a + b, 0);
+
+    return summator;
```

Comment:
The 'summator' function should return a new function that can take the next argument or return the result when called without arguments. The current implementation will not work correctly because it expects an array of arguments instead of a single argument at a time.